kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/parquet/ParquetSuiteBase.scala

```diff
@@ -0,0 +1,266 @@
+    val files = parquetFiles(path)
+
+    var rowCount = 0L
+    files.foreach { file =>
+      // read parquet file using spark and count.
+      rowCount = rowCount + spark.read.parquet(file).count()
+    }
+
+    rowCount
+  }
+
+  def parquetFiles(path: String): Seq[String] = {
+    Files.list(Paths.get(path))
+      .iterator().asScala
+      .map(_.toString)
+      .filter(path => path.endsWith(".parquet"))
+      .toSeq
+  }
+
+  def footer(path: String): ParquetMetadata = {
+    try {
+      ParquetFileReader.readFooter(configuration, new Path(path))
+    } catch {
+      case NonFatal(e) => fail(s"Failed to read footer for file: $path", e)
+    }
+  }
+
+  def tableSchema(path: String): StructType = {
+    Table.forPath(defaultTableClient, path)
+      .getLatestSnapshot(defaultTableClient)
+      .getSchema(defaultTableClient)
+  }
+}
+
+object ParquetSuiteBase {
+  // Parquet file containing data of all supported types and variations
+  val ALL_TYPES_DATA = goldenTableFile("parquet-all-types").toString
+  // Schema of the data in `ALL_TYPES_DATA`
+  val ALL_TYPES_FILE_SCHEMA = new StructType()
+    .add("byteType", ByteType.BYTE)
+    .add("shortType", ShortType.SHORT)
+    .add("integerType", IntegerType.INTEGER)
+    .add("longType", LongType.LONG)
+    .add("floatType", FloatType.FLOAT)
+    .add("doubleType", DoubleType.DOUBLE)
+    .add("decimal", new DecimalType(10, 2))
+    .add("booleanType", BooleanType.BOOLEAN)
+    .add("stringType", StringType.STRING)
+    .add("binaryType", BinaryType.BINARY)
+    .add("dateType", DateType.DATE)
+    .add("timestampType", TimestampType.TIMESTAMP)
+    .add("nested_struct",
+      new StructType()
+        .add("aa", StringType.STRING)
+        .add("ac",
+          new StructType()
+            .add("aca", IntegerType.INTEGER)))
+    .add("array_of_prims", new ArrayType(IntegerType.INTEGER, true))
+    .add("array_of_arrays", new ArrayType(new ArrayType(IntegerType.INTEGER, true), true))
+    .add("array_of_structs",
+      new ArrayType(
+        new StructType()
+          .add("ab", LongType.LONG), true))
+    .add("map_of_prims", new MapType(IntegerType.INTEGER, LongType.LONG, true))
+    .add("map_of_rows",
+      new MapType(
+        IntegerType.INTEGER,
+        new StructType().add("ab", LongType.LONG),
+        true))
+    .add("map_of_arrays",
+      new MapType(
+        LongType.LONG,
+        new ArrayType(IntegerType.INTEGER, true),
+        true))
+
+  // Parquet file containing all variations (int, long and fixed binary) Decimal type data
+  val DECIMAL_TYPES_DATA = goldenTableFile("parquet-decimal-type").toString
+  // Schema of the data in `DECIMAL_TYPES_DATA`
+  val DECIMAL_TYPES_FILE_SCHEMA = new StructType()
+    .add("id", IntegerType.INTEGER)
+    .add("col1", new DecimalType(5, 1)) // stored as int
+    .add("col2", new DecimalType(10, 5)) // stored as long
+    .add("col3", new DecimalType(20, 5)) // stored as fixed binary
```

Comment:
this was removed in the latest PR on master.

Reply:
> this was removed in the latest PR on master.

@vkorukanti sorry, I don't understand. What I need to do here?

Reply:
Please rebase this PR on top of master. I removed these in one of the latest commits on the `master` branch.

Reply:
> Please rebase this PR on top of master. I removed these in one of the latest commits on the `master` branch.

@vkorukanti oh my bad, I forgot to sync my forked repo. It's resolved now. Please review it again, TYSM.